aiomysql/connection.py

```diff
@@ -786,53 +804,196 @@ def _get_server_information(self):
             server_end = data.find(b'\0', i)
             if server_end < 0:  # pragma: no cover - very specific upstream bug
                 # not found \0 and last field so take it all
-                self._auth_plugin_name = data[i:].decode('latin1')
+                server_auth = data[i:].decode('latin1')
             else:
-                self._auth_plugin_name = data[i:server_end].decode('latin1')
+                server_auth = data[i:server_end].decode('latin1')
 
-    def get_transaction_status(self):
-        return bool(self.server_status & SERVER_STATUS.SERVER_STATUS_IN_TRANS)
+            self._server_requested_auth_plugin = server_auth
 
-    def get_server_info(self):
-        return self.server_version
+    async def request_authentication(self):
+        # https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::HandshakeResponse
+        if int(self.server_version.split('.', 1)[0]) >= 5:
+            self.client_flag |= CLIENT.MULTI_RESULTS
 
-    # Just to always have consistent errors 2 helpers
+        if self._user is None:
+            raise ValueError("Did not specify a username")
 
-    def _close_on_cancel(self):
-        self.close()
-        self._close_reason = "Cancelled during execution"
+        # SSL
+        if self._tls_context:
+            # capablities, max packet, charset
+            data = struct.pack('<IIB', self.client_flag, 16777216, 33)
+            data += b'\x00' * (32 - len(data))
+
+            await self.write_packet(data, no_ack=True)
+
+            self._tls_protocol = sslproto.SSLProtocol(self.loop,
+                                                      self,
+                                                      self._tls_context,
+                                                      None,
+                                                      server_side=False)
+            self._original_transport = self.transport
+            self._original_transport._protocol = self._tls_protocol
```

Comment:
This is not good, accessing privet variables very fragile and this property is not available tin uvloop

Reply:
Agreed, if there's a nicer way I'm all ears :D. I looked at how aiosmtpd dealt with the STARTTLS command, which does the same switch to TLS mid connection and they do the same thing.